### PR TITLE
FE-1081 - Update <Banner/> button colors for Hibana

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-1081](https://jira.int.messagesystems.com/browse/FE-1081)

### What Changed
- Added new `Button` variant - `monchrome-primary`
- Updated `ApiErrorBanner` to conditionally handle child button variants
- Added supporting documentation for UX pattern *and* for missing `Button` variants
- Replaced several instances of `Button` being used inside `Banner` components when `ExternalLink` was preferred, improving UX consistency
- ...I updated a totally unrelated page with some one-off props that could be cleanly replaced with `PageDescription` - this has nothing to do with the a/c but I just had to fix it.

### How To Test
- Verify all `Button` instances within `Banner` instances meet the required a/c

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/trackingDomains/components/UnverifiedBanner.js [hunk 1]
```diff
@@ -23,9 +24,7 @@ const UnverifiedBanner = ({ unverifiedDomains, cname }) => {
           with the value of <strong>{cname}</strong>.
         </p>
         <ButtonWrapper>
-          <Button outline external to={LINKS.DOMAIN_VERIFICATION}>
-            Learn more
-          </Button>
+          <ExternalLink to={LINKS.DOMAIN_VERIFICATION}>Learn more</ExternalLink>
```
Comment (bkemper) [inline src/pages/trackingDomains/components/UnverifiedBanner.js:27]: 👏

File: src/pages/trackingDomains/components/UnverifiedBanner.js [hunk 2]
```diff
@@ -23,9 +24,7 @@ const UnverifiedBanner = ({ unverifiedDomains, cname }) => {
           with the value of <strong>{cname}</strong>.
         </p>
         <ButtonWrapper>
```
Comment (bkemper) [inline src/pages/trackingDomains/components/UnverifiedBanner.js:26]: Is ButtonWrapper the right wrapper component for links?

File: src/components/matchbox/Button.stories.mdx [hunk 3]
```diff
@@ -44,6 +45,31 @@ Use the `<Button/>` component for interactive elements that draw the user's atte
...
+</Preview>
+
+### Connected Variant
+
+<Preview>
+  <TextField
+    id="example-text-field"
+    name="example-text-field"
+    label="Example Text Field"
+    connectRight={<Button variant="connected">Clear</Button>}
```
Comment (bkemper) [inline src/components/matchbox/Button.stories.mdx:67]: 🙏

**Comment by bkemper** [comment]
Should this only have an outline?

Screen Shot 2020-05-27 at 3 35 42 PM

**Comment by bkemper** [comment]
@nicklemmon Unit tests are failing.

**Comment by nicklemmon** [comment]
> @nicklemmon Unit tests are failing.

😢 Believe me I know. Snapshot test after snapshot test - nothing functionally breaking, but the tests fail because of a visual redesign. No good. I think we should potentially add an ESLint rule to warn people away from using snapshot tests.

File: src/components/buttonWrapper/ButtonWrapper.js [hunk 4]
```diff
@@ -1,23 +1,14 @@
...
+import styles from './ButtonWrapper.module.scss';
 
-function OGButtonWrapper({ children, marginTop }) {
-  return (
-    <div className={styles.ButtonWrapper} style={{ marginTop }}>
-      {children}
-    </div>
-  );
+function OGButtonWrapper({ children }) {
+  return <div className={styles.ButtonWrapper}>{children}</div>;
```
Comment (bkemper) [inline src/components/buttonWrapper/ButtonWrapper.js:7]: This looks like another candidate to be dropped with OG?

File: src/components/matchbox/Banner.js [hunk 5]
```diff
@@ -1,15 +1,67 @@
...
 }
+
+function Actions(props) {
+  const styles = useHibanaOverride(OGStyles, hibanaStyles); // TODO: Remove and replace styles with <Box/> component when OG theme is removed
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  const { children } = props;
+
+  const renderChildren = () => {
+    const childrenArr = React.Children.map(children, child => child);
```
Comment (bkemper) [inline src/components/matchbox/Banner.js:32]: Isn't this the same as `toArray` — https://reactjs.org/docs/react-api.html#reactchildrentoarray

File: src/components/matchbox/Banner.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  should be gray with an outline
+
+**NOTE:** The following examples only work visually with Hibana enabled.
+
+### With Two Buttons
+
+<Preview>
+  <Banner status="warning" title="This is a warning banner with buttons">
+    <p>This is some content in a banner.</p>
+    <ButtonWrapper>
```
Comment (bkemper) [inline src/components/matchbox/Banner.stories.mdx:28]: `<Banner.Actions>`?

**Comment by bkemper** [comment]
I wasn't expecting this many changes.  I left a couple comments and I'll finish my review a little later.

File: src/components/matchbox/Banner.js (hunk 5 above)
Comment (nicklemmon) [reply src/components/matchbox/Banner.js:32]: Nice. I didn't know about this one.

File: src/components/datePicker/DatePicker.js [hunk 7]
```diff
@@ -309,7 +316,7 @@ export class DatePickerClassComponent extends Component {
           />
         )}
 
-        <ButtonWrapper>
+        <Box mt="500">
```
Comment (bkemper) [inline src/components/datePicker/DatePicker.js:319]: The spacing between buttons was lost in OG.

Screen Shot 2020-05-29 at 12 53 20 PM

File: src/pages/ipPools/ListPage.js [hunk 8]
```diff
@@ -87,17 +87,22 @@ export class IpPoolsList extends Component {
...
       <p>
         In order to establish a positive sending reputation, warm up new dedicated IP addresses by
         gradually sending more emails.
       </p>
-      <ButtonWrapper>
-        <ExternalLink as={Button} outline={true} to={LINKS.IP_WARM_UP}>
-          {'Read our IP Warm-up Overview'}
+
+      <Banner.Actions>
+        <ExternalLink as={Button} to={LINKS.IP_WARM_UP}>
```
Comment (bkemper) [inline src/pages/ipPools/ListPage.js:102]: Should this be as a button?

File: src/components/datePicker/DatePicker.js (hunk 7 above)
Comment (nicklemmon) [reply src/components/datePicker/DatePicker.js:319]: Let me fix this

File: src/pages/ipPools/ListPage.js (hunk 8 above)
Comment (nicklemmon) [reply src/pages/ipPools/ListPage.js:102]: I think so actually...I marked it up as a link originally and it looked pretty weak. I believe this is supposed to be a pretty strong call to action, so I think my mind has been changed about whether a `<Button/>` should be used as a link in certain scenarios inside a `<Banner/>`.

**Pull request merged**